Fix: Updated SkullMetaMock to support PlayerProfile

Nirbose opened the pull request:
# Description
<!-- State the changes of the pull request below. -->
Adds a playerProfile field of type
com.destroystokyo.paper.profile.PlayerProfile to the SkullMetaMock class to store player profile information associated with the skull. In the constructor, the PlayerProfile is now also initialized from the data of the original SkullMeta.
Adds methods to set and retrieve the PlayerProfile associated with the SkullMeta.
The setPlayerProfile and getPlayerProfile methods are now functional, allowing for proper management of the player profile associated with the skull.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

Nirbose committed:
Fix: Updated SkullMetaMock to support PlayerProfile

Adds a playerProfile field of type
com.destroystokyo.paper.profile.PlayerProfile to the SkullMetaMock class
to store player profile information associated with the skull.
In the constructor, the PlayerProfile is now also initialized from the
data of the original SkullMeta.
Adds methods to set and retrieve the PlayerProfile associated with the
SkullMeta.
The setPlayerProfile and getPlayerProfile methods are now functional,
allowing for proper management of the player profile associated with the
skull.

thelooter commented on src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:
```diff
@@ -2613,15 +2615,13 @@ public boolean hasResourcePack()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		this.playerProfile = profile;
```
```suggestion
        Preconditions.checkNotNull(profile, "Profile cannot be null")
		this.playerProfile = profile;
```

And please add a Test for it

thelooter commented on src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:
```diff
@@ -2613,15 +2615,13 @@ public boolean hasResourcePack()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		this.playerProfile = profile;
```
```suggestion
        Preconditions.checkNotNull(profile, "Profile cannot be null");
		this.playerProfile = profile;
```

And please add a Test for it

Nirbose replied:
Add to commit bbe7a424669c78c0fd91e520ec7848bc6a552502

thelooter reviewed (changes requested):
Small Change

Nirbose committed:
Add null check when setting PlayerProfile

This commit introduces an enhancement to the PlayerMock class by adding
a null check when setting the PlayerProfile. The change ensures that the
PlayerProfile argument provided is not null, enhancing the robustness of
the code and preventing potential issues related to a null player
profile. This improvement contributes to the overall reliability and
stability of the codebase.
